definitions/reports/image_mode_hosts.rb

```diff
@@ -0,0 +1,27 @@
+module Reports
+  class ImageModeHosts < ForemanMaintain::Report
+    metadata do
+      description 'Report number of image mode hosts registered by operating system'
+      confine do
+        feature(:katello)
+      end
+    end
+
+    def run
+      merge_data('image_mode_hosts_by_os_count') { image_mode_hosts_by_os_count }
+    end
+
+    # OS usage on image mode hosts
+    def image_mode_hosts_by_os_count
+      query(
+        <<-SQL
+          select max(operatingsystems.name) as os_name, count(*) as hosts_count
```

Comment:
A couple of questions:
Are those name constant across Satellites?
How many names we expect here?
Can we have two os instances with the same name (with a different version perhaps)?

I am worried about the amount of records this can generate in the report. For now we agreed that we do not create nodes based on fields with unbound value counts.

Reply:
> Are those name constant across Satellites?
I am not sure I understand the question, but the name of OS should hopefully not depend on Satellite.

> How many names we expect here?
I'd say that depends on individual deployments, but I wouldn't expect too many. For example, I think that all RHEL systems should be grouped under RedHat. It is less of a name and more family name (I think? I copied the behavior for the existing host_count_by_os). 

> Can we have two os instances with the same name (with a different version perhaps)?
I am fairly certain that those are grouped under the same name.

Reply:
> I am not sure I understand the question, but the name of OS should hopefully not depend on Satellite.
If a user can change the name - we will have different names for different instances, making aggregation impossible. Since you're following the same scheme as `host_count_by_os`, I suppose we already decided that it was good enough.

@adamruzicka  It actually makes me wonder if we don't have an error in `host_count_by_os` - we're grouping by os_id there, but labeling by name. If we have two os's with the same name, but different ids, we will be in trouble.

Reply:
> It actually makes me wonder if we don't have an error in host_count_by_os


It seems we do https://github.com/theforeman/foreman_maintain/pull/997

Reply:
Thanks, I applied the same fix here.